postgres/src/rows.rs

```diff
@@ -259,13 +256,21 @@ impl<'a> Row<'a> {
 pub trait RowIndex {
     /// Returns the index of the appropriate column, or `None` if no such
     /// column exists.
-    fn idx(&self, stmt: &Statement) -> Option<usize>;
+    fn idx<I: AsRef<StatementInfo>>(&self, info: &I) -> Option<usize> {
```

Comment:
It's all about the degree of broken-ness - the odds that anyone has code that implements `RowIndex` or calls `idx` manually are I believe very low, but the odds that anyone refers to `Rows<'a>` with a lifetime seems much higher. We can either cut a new major version or just make the RowIndex change and revert if anyone yells.